### PR TITLE
Moving to JS random

### PR DESCRIPTION
**Pull request by pelikhan**
https://github.com/Microsoft/pxt/pull/2337

### CODE REVIEW

**Comment by mmoskal** [comment]
I think you should get two random ints and use bits from both for randomness. You need 53 bits. Otherwise, some of the bits are not really random, and I think it may show in some situations.

For random range with ints it would be faster to use device.random directly.

**Comment by pelikhan** [comment]
> For random range with ints it would be faster to use device.random directly.

Hummm I tought i was already doing that.

**Comment by mmoskal** [comment]
whoops, sorry. But the 53 bits would be nice.

File: libs/core/core.cpp [hunk 1]
```diff
@@ -467,15 +467,38 @@ TNumber atan2(TNumber y, TNumber x) {
...
+        double maxd = toDouble(max);
+        if (mind > maxd) {
+            double temp = mind;
+            mind = maxd;
+            maxd = temp;
+        }
+        if (maxd == mind)
+            return fromDouble(mind);
+        else
+            return fromDouble(mind + device.random(maxd - mind));
```
Comment (mmoskal) [inline libs/core/core.cpp]: oh, for example for randomRange(0.5, 3.5) this will return 0.5, 1.5, 2.5 or 3.5, but never say 1.83434.

**Pull request merged**